FactorioWebInterface/Data/DefaultAdminAccount.cs

```diff
@@ -0,0 +1,147 @@
+            }
+            
+            if (userCount > 0 && rootCount > 0)
+            {
+                return AccountsNumbers.MultipleAccounts;
+            }
+
+            return AccountsNumbers.NoAccounts;
+        }
+
+        private async Task ClearDefaultUserAsync(string id, bool force = false)
+        {
+            ApplicationUser userResult = await _userManager.FindByIdAsync(id);
+            if (await ValidateDefaultUserAsync(userResult) && !force)
+            {
+                Log.Information("Valid " + Constants.DefaultAdminAccount + " already exists");
+                return;
+            }
+            
+            if (userResult != null)
+            {
+                var deleteResult = await _userManager.DeleteAsync(userResult);
+                if (!deleteResult.Succeeded)
+                {
+                    Log.Error(Constants.DefaultAdminAccount + " couldn't be deleted! This may pose a security risk for your application. Will attempt to delete again at next reboot");
+                }
+                Log.Information(Constants.DefaultAdminAccount + " deleted");
+            }
+        }
+
+        private async Task<bool> ValidateDefaultUserAsync(ApplicationUser user)
+        {
+            if (user == null || user.UserName != _userName)
+            {
+                return false;
+            }
+
+            if (!await _userManager.HasPasswordAsync(user))
+            {
+                return false;
+            }
+
+            if (!await _userManager.IsInRoleAsync(user, Constants.RootRole))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private async Task CreateDefaultUserAsync(string id)
+        {
+            ApplicationUser user = new ApplicationUser()
+            {
+                Id = id,
+                UserName = _userName
+            };
+
+            var result = await _userManager.CreateAsync(user);
+            if (!result.Succeeded)
+            {
+                Log.Error("Couldn't create " + Constants.DefaultAdminAccount);
+                return;
+            }
+
+            result = await _userManager.AddToRoleAsync(user, Constants.RootRole);
+            if (!result.Succeeded)
+            {
+                Log.Error("Couldn't add role to " + Constants.DefaultAdminAccount);
+                return;
+            }
+            string password = Guid.NewGuid().ToString();
+            result = await _userManager.AddPasswordAsync(user, password);
+            if (!result.Succeeded)
+            {
+                Log.Error("Couldn't add password to " + Constants.DefaultAdminAccount);
+                return;
+            }
+            Log.Warning(Constants.DefaultAdminAccount + " named \'" + _userName + "\' created with the password: " + password + "\nThis action potential exposes your interface, creating a new account and restarting this web interface will disable the default admin account");
```

Comment:
I know I suggested logging, but I'm having second thoughts. Maybe outputting to file would be better. It would be easier to find and also allow delete the file. Thoughts?